### PR TITLE
Reusable Configurations topic

### PR DESCRIPTION
**Pull request by mcherem**
Adding new file for config reuse feature

### CODE REVIEW

**Comment by AlfaroMiguel** [comment]
Looks good to me, but maybe we should add that the user is the only one who can change his own connection and nobody else. If someone else see the connection it won't see the three dots in order to edit/delete the connection. What do you think about @mcherem ?

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
+= Create a Connection for Reuse by Yourself or Others
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords:
+
+:imagesdir: ../assets/images
+
+You can create, change and reuse a connection in a Flow Designer project instead of adding your authentication information manually. Once a reusable connection is created, it is available for all Flow Designer projects.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: Suggest deleting "create, change and".  "Reuse" is the new feature that we want to call attention to.

Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: suggest changing the end to add clarity: "for all Flow Designer projects in your business group."

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+You can reuse connections:
+
+* If you are a system admin, you can enable non-admins to connect to your system using administrative authentication. This is important if the integration requires administrative system permissions.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: I have never heard the term "administrative authentication".  I think people ordinarily use "administrative credentials" or "an administrative connection."

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+* If you are responsible for maintaining the connection, you can ensure that the integration is unaffected when the user who created the Flow Designer project changes their personal password or leaves the organization.
+
+== Create a Reusable Connection
+
+Adding a reusable connection is available to you when configuring a connector in Flow Designer.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: The "available to you" language seems awkward to me.  I suggest "You can add a reusable connection when configuring a connector..." or "Add a connection when configuring..."

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+. Once you are viewing the Flow Designer canvas, click *+* to add a card.  A list of the available connectors displays within the card.
+
+. In the Search field, type the name of the connector and select it. The connection section of the connector displays a list of operations that you can use to trigger Mule applications with the connector.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: Do you think this whole thing can be shortened to "Choose an operation."?

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+. To make this connection available for users in your business group, leave the *share this connection with my business group* checkbox checked. To maintain security, only you as the creator of a shared connection can view your authentication credentials.
+
+. If you don’t need to share this connection with other users, uncheck the box.
+
+. Select a connection type and fill in the necessary information.
+
+. Validate that this connection works by clicking *Test*.
+
+. In order to save your connection, you’ll need to click *Save* twice.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: I suggest rewording so it sounds less like a bug.  I suggest "Click save in the connection section, and click save again in the reusable configuration section."

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+. Select the appropriate radio button for the connection you intend to use.
+
+. Click *Test* to make sure that the credentials are valid.
+
+. Click *Save* after the test succeeds.
+
+== Change a reusable connection
+
+When your credentials change, you can edit or delete the shared connection you created so that all apps dependent upon it continue to work.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: Technically, it should be "so that all projects" instead of "so that all apps".  A Flow Designer project only becomes an app once it is pushed to production or exported.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+. Select the connector and click the three-dots icon under Reusable Configurations. If you have the connector open, select a connector under Connection.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: I suggest clarifying the clicks, perhaps with the following language: "Open the reusable configuration section of your connector.  You see a list of connections.  Select one and click the three-dots icon."

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+. If you want to remove the connection, select *Delete*. You must have another connection available to use in order to delete your reusable connection.
+
+. If you want to edit the connection, select *Edit* and make your changes.
+
+. To have your changes take effect, click *Save* twice.
```
Comment (ryanhol) [inline modules/ROOT/pages/fd-reuse-config.adoc]: Instead of "click *Save* twice" I recommend "after saving your connection, click save in the reusable configuration section of the connector."  It makes it seem less like a bug.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 1 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 2 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: I was using the term authentication because it looks like it is used throughout most of the FD docs. What about 'your authentication'?

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 3 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 4 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Ha! Yeah, that's a win in the minimalist category.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 5 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: I think this is being fixed, right?

Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: I have added your suggested change.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 6 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Thanks for this. I see why it's incorrect in this instance.

Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 7 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: I agree, much better. Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 8 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+. Select the appropriate radio button for the connection you intend to use.
+
+. Click *Test* to make sure that the credentials are valid.
+
+. Click *Save* after the test succeeds.
+
+== Change a reusable connection
+
+When your credentials change, you can edit or delete the shared connection you created so that all projects dependent upon it continue to work.
```
Comment (kbostrom) [inline modules/ROOT/pages/fd-reuse-config.adoc]:
```suggestion
When your credentials change, you can edit or delete the shared connection you created so that all projects dependent upon it continue to work:
```

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+. Open the reusable configuration section of your connector. You see a list of connections. Select one and click the three-dots icon.
```
Comment (kbostrom) [inline modules/ROOT/pages/fd-reuse-config.adoc]:
```suggestion
. Open the reusable configuration section of your connector. 
. In the list of connections, select one and click the ... icon.
```
There are several names for the icons that have three horizontal or three vertical dots.
If they are horizontal, you could call this an "ellipsis" icon. 
I've also heard it called a Twinkie!
Some people call three vertical dots the "hamburger" icon.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+. If you want to remove the connection, select *Delete*. You must have another connection available to use in order to delete your reusable connection.
```
Comment (kbostrom) [inline modules/ROOT/pages/fd-reuse-config.adoc]:
```suggestion
. If you want to remove the connection, select *Delete*. 
You must have another connection available before you can delete your reusable connection.
```

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+. If you want to edit the connection, select *Edit* and make your changes.
+
+. Click *Save* in the connection section, and click *Save* again in the reusable configuration section.
+
+This immediately updates all Flow Designer projects being designed, but not Mule runtime apps already deployed to CloudHub.
```
Comment (kbostrom) [inline modules/ROOT/pages/fd-reuse-config.adoc]:
```suggestion
Completing this task immediately updates all Flow Designer projects being designed; it does not update Mule runtime apps already deployed to CloudHub.
```

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+:imagesdir: ../assets/images
+
+You can reuse a connection in a Flow Designer project instead of adding your authentication information manually. Once a reusable connection is created, it is available for all Flow Designer projects in your business group.
+
+You can reuse connections:
+
+* If you are a system admin, you can enable non-admins to connect to your system using your authentication. This is important if the integration requires administrative system permissions.
+
+* If you are responsible for maintaining the connection, you can ensure that the integration is unaffected when the user who created the Flow Designer project changes their personal password or leaves the organization.
+
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Added a sentence.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+== Create a Reusable Connection
+
+Add a reusable connection when configuring a connector in Flow Designer.
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Nope, removing.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+. Choose an operation.
+
+. Click *Add connection*.
+
+. Enter a unique and descriptive name that your users can easily recognize.
+
+. To make this connection available for users in your business group, leave the *share this connection with my business group* checkbox checked. To maintain security, only you as the creator of a shared connection can view your authentication credentials.
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+. If you don’t need to share this connection with other users, uncheck the box.
+
+. Select a connection type and fill in the necessary information.
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+. Validate that this connection works by clicking *Test*.
+
+. Click *Save* in the connection section, and click *Save* again in the reusable configuration section.
+
+== Use a Reusable Connection Created in Another Project
+
+Once a reusable connection is created, it is available for all Flow Designer projects.
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+
+. Choose a connector and operation.
+
+. View a list of reusable connections. To help you choose the right connection the date the reusable connector was created is visible under the connection.
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed. The lead-in sentence says that once a reuseable connection is created, then all the reusable ones are viewable. It's literally the first thing the user sees.

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+
+. Select the appropriate radio button for the connection you intend to use.
+
+. Click *Test* to make sure that the credentials are valid.
+
+. Click *Save* after the test succeeds.
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Would it be more accurate to say, click save when the test succeeds?

File: modules/ROOT/pages/fd-reuse-config.adoc [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+
+== Change a reusable connection
```
[the remark this answers is not in the document]
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: But I always thought that change implies add, edit, and delete where edit only implies change existing.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 9 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 10 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: would love a standard name.  I am using what Robert used in FD to be consistent.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 11 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.

File: modules/ROOT/pages/fd-reuse-config.adoc (hunk 12 above)
Comment (mcherem) [reply modules/ROOT/pages/fd-reuse-config.adoc]: Fixed.
